oxide.go

```diff
@@ -593,3 +688,46 @@ func toRancherMachineState(instanceState oxide.InstanceState) state.State {
 		return state.None
 	}
 }
+
+// AdditionalDisk represents a disk attached to an instance.
+type AdditionalDisk struct {
+	// Required. The size of the disk in bytes.
+	Size uint64
+
+	// An optional label to use in the disk name for ease of identification.
+	Label string
+}
+
+func ParseAdditionalDisk(s string) (AdditionalDisk, error) {
+	var sizeStr string
+	label := "additional"
+
+	fields := strings.Split(s, ",")
+	switch len(fields) {
+	case 2:
+		sizeStr = fields[0]
+		if fields[1] != "" {
+			label = fields[1]
+		}
+	case 1:
+		sizeStr = fields[0]
+	default:
+		return AdditionalDisk{}, fmt.Errorf("invalid format %q, expected size[,label]", s)
+	}
+
+	size, err := humanize.ParseBytes(sizeStr)
+	if err != nil {
+		return AdditionalDisk{}, fmt.Errorf("failed parsing size %q %w", sizeStr, err)
+	}
+
+	a := AdditionalDisk{
+		Size:  size,
+		Label: label,
+	}
+
+	return a, nil
+}
+
+func (a AdditionalDisk) Name(machineName string, diskNumber int) string {
+	return fmt.Sprintf("disk-%02d-%s-%s", diskNumber, a.Label, machineName)
```

Comment:
@one-horned-flying this will update disk naming with leading zeroes (e.g., `0` -> `00`, `1` -> `01`).